src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -158,34 +159,47 @@ def safe_for_comment(
                 tags={"type": GithubAPIErrorType.UNKNOWN.value, "code": e.code},
             )
             logger.exception("github.open_pr_comment.unknown_api_error", extra={"error": str(e)})
-        return False
+        return False, []
 
     safe_to_comment = True
-    if pullrequest_resp["state"] != "open":
-        metrics.incr(
-            OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "incorrect_state"}
-        )
-        safe_to_comment = False
-    if pullrequest_resp["changed_files"] > OPEN_PR_MAX_FILES_CHANGED:
+
+    changed_file_count = 0
+    changed_lines_count = 0
+
+    for file in pr_files:
+        filename = file["filename"]
+        # don't count the file if it was added or is not a Python file
+        if file["status"] == "added" or not filename.endswith(".py"):
+            continue
+
+        changed_file_count += 1
+        changed_lines_count += file["changes"]
+
+    if changed_file_count > OPEN_PR_MAX_FILES_CHANGED:
         metrics.incr(
             OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "too_many_files"}
         )
         safe_to_comment = False
-    if pullrequest_resp["additions"] + pullrequest_resp["deletions"] > OPEN_PR_MAX_LINES_CHANGED:
+    if changed_lines_count > OPEN_PR_MAX_LINES_CHANGED:
         metrics.incr(
             OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "too_many_lines"}
         )
         safe_to_comment = False
-    return safe_to_comment
 
+    if not safe_to_comment:
+        pr_files = []
+
+    return safe_to_comment, pr_files
```

Comment:
i chose to return the response and use it in a different function rather than do everything in this function (determine whether the PR is safe for comment + get pr filenames + get pr diffs) to avoid overloading it